oap-server/server-receiver-plugin/skywalking-telegraf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/telegraf/provider/config/TelegrafConfigs.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.receiver.telegraf.provider.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.oap.server.library.module.ModuleStartException;
+import org.apache.skywalking.oap.server.library.util.CollectionUtils;
+import org.apache.skywalking.oap.server.library.util.ResourceUtils;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class TelegrafConfigs {
+    public static List<TelegrafConfig> loadConfigs(String path, List<String> fileNames) throws ModuleStartException {
+        if (CollectionUtils.isEmpty(fileNames)) {
+            return Collections.emptyList();
+        }
+
+        File[] configs;
+        try {
+            configs = ResourceUtils.getPathFiles(path);
+        } catch (FileNotFoundException e) {
+            throw new ModuleStartException("Load telegraf configs failed", e);
+        }
+
+        return Arrays.stream(configs)
+                .filter(File::isFile)
+                .map(f -> {
+                    String fileName = f.getName();
+                    int dotIndex = fileName.lastIndexOf('.');
+                    fileName = (dotIndex == -1) ? fileName : fileName.substring(0, dotIndex);
+                    if (!fileNames.contains(fileName)) {
+                        return null;
+                    }
+                    try (Reader r = new FileReader(f)) {
+                        return new Yaml().loadAs(r, TelegrafConfig.class);
+                    } catch (IOException e) {
+                        log.warn("Reading file {} failed", f, e);
```

Comment:
Please throw module start exception, this is a fatal error

Reply:
> Please throw module start exception, this is a fatal error

I set the `loadConfigs` method throws the `ModuleStartException` , should I change it?
![image](https://user-images.githubusercontent.com/79000735/190843723-2cbbfe34-9552-424c-a370-28d31f35f2c4.png)

Reply:
> > Please throw module start exception, this is a fatal error
> 
> I set the `loadConfigs` method throws the `ModuleStartException` , should I change it? ![image](https://user-images.githubusercontent.com/79000735/190843723-2cbbfe34-9552-424c-a370-28d31f35f2c4.png)

I mean you should throw a `ModuleStartException` here, logging it is not enough.

```suggestion
                        throw new ModuleStartException(e);
```